### ISSUE TITLE
discussion: accept scorer rather than metric + response function?

### ISSUE THREAD

**Issue by jeromedockes** [mind-inria/hidimstat#787]
scikit-learn interfaces that need to know how to evaluate an estimator typically accept a 'scoring' parameter: 

https://scikit-learn.org/stable/modules/model_evaluation.html#the-scoring-parameter-defining-model-evaluation-rules

basically a function `f(estimator, X_test, y_test) -> score`.

this is different from say, CFI which expects a 'method', e.g. predict_proba (what scikit-learn calls the 'response method') and a 'loss', e.g. mean_squared_error (what scikit-learn calls a 'metric').

accepting 'scoring' could be a more familiar interface for users, and avoid needing to specify 2 parameters when we use one of the scikit-learn built-in functions, because for example scikit-learn knows the response function for mse is predict()

scoring can be turned into a scorer with [check_scoring](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.check_scoring.html) eg 

```
>>> check_scoring(Ridge(), 'neg_mean_squared_error')
make_scorer(mean_squared_error, greater_is_better=False, response_method='predict')
```

a user who has the loss/metric and response method, they can build a scorer with [make_scorer](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.make_scorer.html)

**Comment by jpaillard** [comment]
Thx, yes, that would clarify things and avoid conflicting arguments. 

Maybe we want to write our own `check_scoring,` however. Indeed, a classifier would typically use an `accuracy` score, while we would rather set the default to `log_loss`. The same regressor would return R2 by default, whereas we would instead want `mse`.

**Comment by GrituX** [comment]
I'm gonna start working on this. This entails a significant revamp of the API, so it's gonna take a bit of time.

So the idea is to start with implementing our own `check_scoring()` method which would `get_scorer` from a given `scoring` input. I'm probably missing something because I can't identify where the response method can be retrieved from the `scorer` in sklearn's API to have that information internally.

**Comment by GrituX** [comment]
For `BasePerturbation`, `PFI`, and `CFI` I managed to do without, so I'll see for the rest.
I'll open PRs progressively so that everything is incremental, and code reviews are not too overwhelming.
